---
title: "2022_01_25_0946-compare_STARstringent_vs_qPCR"
author: "Ming"
date: "`r format(Sys.time(), '%Y_%m_%d_%H_%M')`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
require(tidyverse)
require(ggplot2)
'%ni%' <- Negate('%in%')
```

# read in data

```{r}
data.dir = "/Users/minghan/bioinfoproj/FAM72_homolog/pipelinesuite_STARstringent_output/"
```

## TPM gene expr

```{r}
TPM_1 = 
  read.table(file = paste0(data.dir, "2022-01-20_FAM72_STARstringent_try1_test_gene_expression_TPM.tsv"), header = T) %>% 
  filter(str_detect(Symbol, "FAM72"))
TPM_2 = 
  read.table(file = paste0(data.dir, "2022-01-24_FAM72_try2_STARstringent_allSamples_gene_expression_TPM.tsv"), header = T) %>% 
  filter(str_detect(Symbol, "FAM72")) %>% 
  select(-X293T_S44_L001)
TPM_2

TPM_gene = 
  left_join(TPM_1, TPM_2, by = c("GeneID", "Symbol")) %>% 
  arrange(Symbol) %>% 
  select(-GeneID)
TPM_gene %>% print.data.frame()
#   Symbol X293T_S44_L001 X293T_S44_L002 A549_S43_L001 A549_S43_L002 Daudi_S41_L001 Daudi_S41_L002 HCT116_S40_L001 HCT116_S40_L002 MDA231_S42_L001 MDA231_S42_L002 SW480_S39_L001 SW480_S39_L002
# 1 FAM72A          13.06          16.51         16.49          4.19          12.87          13.80           14.76           20.81           16.88           26.99          18.33          24.35
# 2 FAM72B          10.41          12.85          5.78          2.10          22.61          13.33            1.96            2.04           32.01           15.11           5.41           3.24
# 3 FAM72C           0.78           0.00          0.76          1.57           1.45           0.75            0.00            0.00            4.00            2.80           0.00           0.00
# 4 FAM72D           5.16           5.31          2.14          2.23           4.78           2.13            2.09            1.44            5.03            3.96           2.21           1.72
TPM_gene.t = 
  as.tibble(t(TPM_gene %>% column_to_rownames("Symbol")), rownames = "Samples")
# TPM_gene.t

write.csv(x = TPM_gene.t, file = paste0(data.dir, "wrangled_data/TPM_gene.FAM72.t.csv"), quote = F, row.names = F)
```

## TPM_for_cbioportal mRNA expr

```{r}
TPM_for_cbioportal_1 = 
  read.table(file = paste0(data.dir, "2022-01-20_FAM72_STARstringent_try1_test_mRNA_expression_TPM_for_cbioportal.tsv"), header = T) %>% 
  filter(str_detect(Symbol, "FAM72"))
TPM_for_cbioportal_1

TPM_for_cbioportal_2 = 
  read.table(file = paste0(data.dir, "2022-01-24_FAM72_try2_STARstringent_allSamples_mRNA_expression_TPM_for_cbioportal.tsv"), header = T) %>% 
  filter(str_detect(Symbol, "FAM72")) %>% 
  select(-X293T_S44_L001)
TPM_for_cbioportal_2

TPM_mRNA = 
  left_join(TPM_for_cbioportal_1, TPM_for_cbioportal_2, by = c("GeneID", "Symbol")) %>% 
  arrange(Symbol) %>% 
  select(-GeneID)
TPM_mRNA %>% print.data.frame()
#   Symbol X293T_S44_L001 X293T_S44_L002 A549_S43_L001 A549_S43_L002 Daudi_S41_L001 Daudi_S41_L002 HCT116_S40_L001 HCT116_S40_L002 MDA231_S42_L001 MDA231_S42_L002 SW480_S39_L001 SW480_S39_L002
# 1 FAM72A       3.707083       4.045268     4.0435195     2.0669502      3.6859401      3.7865964       3.8836208       4.3792051        4.077243        4.754353       4.196135      4.6058499
# 2 FAM72B       3.379898       3.683696     2.5310695     1.0703893      4.4988891      3.7366049       0.9708537       1.0285692        5.000451        3.917432       2.435629      1.6959938
# 3 FAM72C      -0.358454             NA    -0.3959287     0.6507646      0.5360529     -0.4150375              NA              NA        2.000000        1.485427             NA             NA
# 4 FAM72D       2.367371       2.408712     1.0976108     1.1570437      2.2570106      1.0908534       1.0635029       0.5260688        2.330558        1.985500       1.144046      0.7824086

TPM_mRNA.t = 
  as.tibble(t(TPM_mRNA %>% column_to_rownames("Symbol")), rownames = "Samples")
# TPM_mRNA.t

write.csv(x = TPM_mRNA.t, file = paste0(data.dir, "wrangled_data//TPM_mRNA.FAM72.t.csv"), quote = F, row.names = F)
```

## FPKM

```{r}
load(paste0(data.dir, "2022-01-20_FAM72_STARstringent_try1_test_rsem_expression_results.RData"))
genes.formatted_1 = genes.formatted
load(paste0(data.dir, "2022-01-24_FAM72_try2_STARstringent_allSamples_rsem_expression_results.RData"))
genes.formatted_2 = genes.formatted

FPKM = 
  left_join(genes.formatted_1$fpkm, genes.formatted_2$fpkm %>% select(-`293T_S44_L001`), by = c("GeneID", "Symbol"))
FPKM

write.csv(x = FPKM, file = paste0(data.dir, "wrangled_data/FPKM.csv"), quote = F, row.names = F)

FPKM.FAM72 =
  FPKM %>% 
  filter(str_detect(Symbol, "FAM72"))

FPKM.FAM72.t =
  as.tibble(t(FPKM.FAM72 %>% column_to_rownames("Symbol") %>% select(-GeneID)), rownames = "Samples")

write.csv(x = FPKM.FAM72.t, file = paste0(data.dir, "wrangled_data/FPKM.FAM72.t.csv"), quote = F, row.names = F)
```

## RSEM expected_count

```{r}
expected_count = 
  left_join(genes.formatted_1$expected_count, genes.formatted_2$expected_count %>% select(-`293T_S44_L001`), by = c("GeneID", "Symbol"))
expected_count

write.csv(x = expected_count, file = paste0(data.dir, "wrangled_data/expected_count.csv"), quote = F, row.names = F)

expected_count.FAM72 =
  expected_count %>% 
  filter(str_detect(Symbol, "FAM72"))

expected_count.FAM72.t =
  as.tibble(t(expected_count.FAM72 %>% column_to_rownames("Symbol") %>% select(-GeneID)), rownames = "Samples")

write.csv(x = expected_count.FAM72.t, file = paste0(data.dir, "wrangled_data/expected_count.FAM72.t.csv"), quote = F, row.names = F)
```

# --------------------------------------------

# compare

## read in qPCR results

```{r}
fam72_qPCR.t = 
  read.csv(file = "/Users/minghan/GDrive_minghanpughlab/PughLabPMH/_projects/homolog_seq/data/fam72_qPCR.t.csv") %>% 
  filter(., apply(., 1, function(a_row) all(!is.na(a_row))))
# fam72_qPCR.t
```

## try TPM

### get replicate mean

```{r}
fam72_qPCR =
  as.tibble(t(fam72_qPCR.t %>% column_to_rownames(var="Samples")), rownames = "Symbol")
fam72_qPCR.mean = 
  fam72_qPCR %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA 231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))

TPM_gene.mean = 
  TPM_gene %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("X293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))

fam72_qPCR.mean.long = 
  fam72_qPCR.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "qPCR_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)

TPM_gene.mean.long = 
  TPM_gene.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "TPMgene_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
```

### compare

#### raw values (good)

```{r}
fam72_qPCR_TPMgene_mean.joined = 
  left_join(fam72_qPCR.mean.long, TPM_gene.mean.long, by=c("Symbol_Cell_line", "Symbol", "Cell_line"))
# fam72_qPCR_TPMgene_mean.joined

library(ggpubr)
```

##### all (R=0.73, p=5.8e-05)

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs qPCR FAM72 homolog expression correlation")

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCR_FAM72expr.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Cell_line'

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs qPCR FAM72 homolog expression correlation\nby cell line")
## p-value non-sig cuz too little samples

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCR_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Symbol' (FAM72D: R=0.63, p=0.18)

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs qPCR FAM72 homolog expression correlation\nby paralog")
## FAM72D not very good...

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCR_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

#### log2 qPCR (bad, 2 outliers)

- some say if qPCR is measuring doubling relative to reference (2^-ddCt method), 
    - value is essentially 'fold change'
    - should log2(fold change) to get it linear

```{r}
fam72_qPCR_TPMgene_mean.joined.log2qPCR = 
  fam72_qPCR_TPMgene_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean))
# fam72_qPCR_TPMgene_mean.joined.log2qPCR

library(ggpubr)

## all
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs log2(qPCR) FAM72 homolog expression correlation")
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCRlog2_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)


## grouped by 'Cell_line'
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs log2(qPCR) FAM72 homolog expression correlation\nby cell line")
## p-value non-sig cuz too little samples
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCRlog2_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Symbol'
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "STAR stringent parameters\nTPM vs log2(qPCR) FAM72 homolog expression correlation\nby paralog")
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMvsQPCRlog2_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

- terrible...

#### log2 both (bad)

- some also log2 both...
- Everaert,2017-Benchmarking of RNA-sequencing analysis workflows using wholetranscriptome RT-qPCR expression data

```{r}
fam72_qPCR_TPMgene_mean.joined.log2both = 
  fam72_qPCR_TPMgene_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean)) %>% 
  mutate(TPMgene_mean_log2 = log2(TPMgene_mean))
fam72_qPCR_TPMgene_mean.joined.log2both

library(ggpubr)

## all
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "STAR stringent parameters\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation")
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMlog2vsQPCRlog2_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Cell_line'
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "STAR stringent parameters\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation\nby cell line")
## p-value non-sig cuz too little samples
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMlog2vsQPCRlog2_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Symbol'
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "STAR stringent parameters\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation\nby paralog")
ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_TPMlog2vsQPCRlog2_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

- a bit better, but still bad...

## -------------------------------------------

## try FPKM

### get replicate mean

```{r}
FPKM.FAM72.mean = 
  FPKM.FAM72 %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))
FPKM.FAM72.mean

FPKM.FAM72.mean.long = 
  FPKM.FAM72.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "FPKMmean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
FPKM.FAM72.mean.long
```

### compare

#### raw values

```{r}
fam72_qPCR_FPKM_mean.joined = 
  left_join(fam72_qPCR.mean.long, FPKM.FAM72.mean.long, by=c("Symbol_Cell_line", "Symbol", "Cell_line"))
fam72_qPCR_FPKM_mean.joined

library(ggpubr)

## all
fam72_qPCR_FPKM_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=FPKMmean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson")
## a little worse than TPM

## grouped by 'Cell_line'
fam72_qPCR_FPKM_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=FPKMmean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson")
## a little worse than TPM

## grouped by 'Symbol'
fam72_qPCR_FPKM_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=FPKMmean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson")
## a little worse than TPM
```

#### log2 qPCR (bad)

```{r}
fam72_qPCR_FPKM_mean.joined.log2qPCR = 
  fam72_qPCR_FPKM_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean))
fam72_qPCR_FPKM_mean.joined.log2qPCR

library(ggpubr)

## all
fam72_qPCR_FPKM_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson")

## grouped by 'Cell_line'
fam72_qPCR_FPKM_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson")

## grouped by 'Symbol'
fam72_qPCR_FPKM_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson")
```

- same as TPM, terrible...

#### log2 both (bad)

- some also log2 both...
- Everaert,2017-Benchmarking of RNA-sequencing analysis workflows using wholetranscriptome RT-qPCR expression data

```{r}
fam72_qPCR_FPKM_mean.joined.log2both = 
  fam72_qPCR_FPKM_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean)) %>% 
  mutate(FPKMmean_log2 = log2(FPKMmean))
fam72_qPCR_FPKM_mean.joined.log2both

library(ggpubr)

## all
fam72_qPCR_FPKM_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson")

## grouped by 'Cell_line'
fam72_qPCR_FPKM_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean_log2, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson")

## grouped by 'Symbol'
fam72_qPCR_FPKM_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=FPKMmean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson")
```

- same as TPM, bad

## -------------------------------------------

## try DESeq2 normalized

- https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
- best for comparing a single gene, across samples

### DESeq2 normalize

```{r}
library(DESeq2)

## wrangle count data into matrix
expected_count.mat = 
  expected_count %>% 
  select(-GeneID) %>% 
  distinct(Symbol, .keep_all=T) %>% 
  column_to_rownames(var="Symbol") %>% 
  as.matrix() %>% 
  round()
expected_count.mat["FAM72A",]
expected_count.mat %>% dim() # [1] 59050    12

## make some fake 'conditions' and 'metadata'
condition = factor(c(rep("cell_line", 12)))
condition
metadata = data.frame(row.names=colnames(expected_count.mat), condition)
metadata

## Create DESeq2Dataset object
dds <- DESeqDataSetFromMatrix(countData = expected_count.mat, 
                              colData = metadata, 
                              design = ~ 1)
# View(counts(dds))

dds <- estimateSizeFactors(dds)
sizeFactors(dds)
#  293T_S44_L001   293T_S44_L002   A549_S43_L001   A549_S43_L002  Daudi_S41_L001  Daudi_S41_L002 
#      1.0389029       1.0113373       0.9041064       0.8652556       1.0142727       0.9752519  
# HCT116_S40_L001 HCT116_S40_L002 MDA231_S42_L001 MDA231_S42_L002  SW480_S39_L001  SW480_S39_L002 
#      0.9063638       0.8738586       1.1850304       1.1217014       1.2500868       1.2033610 

options(scipen = 999)   ## decimal

DESeq2_normalized_counts <- counts(dds, normalized=TRUE)
DESeq2_normalized_counts

write.table(DESeq2_normalized_counts,
            file=paste0(data.dir,"wrangled_data/DESeq2_normalized_counts.txt"),
            sep="\t", quote=F, col.names=NA)

DESeq2_normalized_counts.FAM72 =
  DESeq2_normalized_counts %>% 
  as.data.frame %>% 
  rownames_to_column(var="Symbol") %>% 
  filter(str_detect(Symbol, "FAM72")) %>% 
  arrange(Symbol)
DESeq2_normalized_counts.FAM72

write.table(DESeq2_normalized_counts.FAM72,
            file=paste0(data.dir,"wrangled_data/DESeq2_normalized_counts.FAM72.txt"),
            sep="\t", quote=F, row.names=F)
```

### get replicate mean

```{r}
DESeq2norm.FAM72.mean = 
  DESeq2_normalized_counts.FAM72 %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))
DESeq2norm.FAM72.mean

DESeq2norm.FAM72.mean.long = 
  DESeq2norm.FAM72.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "DESeq2norm_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
DESeq2norm.FAM72.mean.long
```

### compare

#### raw values

```{r}
fam72_qPCR_DESeq2norm_mean.joined = 
  left_join(fam72_qPCR.mean.long, DESeq2norm.FAM72.mean.long, by=c("Symbol_Cell_line", "Symbol", "Cell_line"))
fam72_qPCR_DESeq2norm_mean.joined

library(ggpubr)
```

##### all (R=0.82, p=8.2e-07)

```{r}
options(scipen = 0)     ## back to sci notation
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "STAR stringent parameters\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation")

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_DESeq2normvsQPCR_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Cell_line'

```{r}
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "STAR stringent parameters\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation\nby cell line")
## p-value non-sig cuz too little samples

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_DESeq2normvsQPCR_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Symbol' (FAM72D: R=0.72, p=0.1)

- suppose to be what DESeq2 is best at normalizing
    - a single gene, across samples

```{r}
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "STAR stringent parameters\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation\nby paralog")
## FAM72A is worse...
## FAM72D is better

ggsave(filename = paste0(data.dir, "figures/corr_STARstringent_DESeq2normvsQPCR_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

<br>

# EOF