---
title: "2022_01_17_1431-compare_kallisto_vs_qPCR.Rmd"
author: "Ming"
date: "`r format(Sys.time(), '%Y_%m_%d_%H_%M')`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
require(tidyverse)
require(ggplot2)
'%ni%' <- Negate('%in%')
```

# read in data

```{r}
data.dir = "/Users/minghan/bioinfoproj/FAM72_homolog/kallisto_output/"
```

## combine samples into 1 df

```{r}
options(scipen = 999)   ## decimal
options(scipen = 0)     ## back to sci notation

samples.vec = sample = gsub("_abundance.tsv", "", list.files(data.dir))
est_count.trans.list = list()
est_count.gene.list = list()
tpm.trans.list = list()
tpm.gene.list = list()

for (sample in samples.vec){
  # print(sample)
  # sample = samples.vec[1]; sample
  df = 
    read.table(file = paste0(data.dir, sample, "_abundance.tsv"), header = T) %>% 
    separate(target_id, into = c("ENST","ENSG","OTTHUMG","OTTHUMT","transcript_id",
                                 "gene_id","length","type","empty"), sep = "\\|") %>% 
    select(-c(OTTHUMG, OTTHUMT, empty))
  # df$gene_id %>% unique() %>% length() # [1] 59050
  # df
  
  ## estimated counts
  est_count.trans.df = df %>% select(-tpm)
  est_count.trans.list[[sample]] = 
    est_count.trans.df %>% rename(!!sample := est_counts)
  
  est_count.gene.df =
    est_count.trans.df %>% 
    select(gene_id, est_counts) %>% 
    group_by(gene_id) %>% 
    summarize(est_counts_gene = sum(est_counts)) %>% 
    ungroup()
  est_count.gene.ENSG.df = 
    inner_join(est_count.gene.df, 
               est_count.trans.df %>% 
                 select(ENSG, gene_id) %>% distinct(gene_id, .keep_all = TRUE), 
               by=c("gene_id")) %>% 
    select(ENSG, gene_id, est_counts_gene)
  est_count.gene.list[[sample]] = 
    est_count.gene.ENSG.df %>% rename(!!sample := est_counts_gene)
  
  ## TPM
  tpm.trans.df = df %>% select(-est_counts)
  tpm.trans.list[[sample]] = 
    tpm.trans.df %>% rename(!!sample := tpm)
  
  tpm.gene.df = 
    tpm.trans.df %>% 
    select(gene_id, tpm) %>% 
    group_by(gene_id) %>% 
    summarize(tpm_gene = sum(tpm)) %>% 
    ungroup()
  tpm.gene.ENSG.df = 
    inner_join(tpm.gene.df, 
               tpm.trans.df %>% 
                 select(ENSG, gene_id) %>% distinct(gene_id, .keep_all = TRUE), 
               by=c("gene_id")) %>% 
    select(ENSG, gene_id, tpm_gene)
  tpm.gene.list[[sample]] = 
    tpm.gene.ENSG.df %>% rename(!!sample := tpm_gene)
}
est_count.trans.list[[1]]
est_count.gene.list[[1]]
tpm.trans.list[[1]]
tpm.gene.list[[1]]
```

### estimated count

```{r}
## transcript level
est_count.trans.df = 
  est_count.trans.list %>% 
  reduce(inner_join, by=c("ENST", "ENSG", "transcript_id", "gene_id", "length", "type")) %>% 
  select(-starts_with("eff_length")) %>% 
  rename("Symbol" = "gene_id")
est_count.trans.df

## gene level
est_count.gene.df = 
  est_count.gene.list %>% 
  reduce(inner_join, by=c("ENSG", "gene_id")) %>% 
  rename("Symbol" = "gene_id")
est_count.gene.df
```

### TPM

```{r}
## transcript level
tpm.trans.df = 
  tpm.trans.list %>% 
  reduce(inner_join, by=c("ENST", "ENSG", "transcript_id", "gene_id", "length", "type")) %>% 
  select(-starts_with("eff_length")) %>% 
  rename("Symbol" = "gene_id")
tpm.trans.df

## gene level
tpm.gene.df =
  tpm.gene.list %>% 
  reduce(inner_join, by=c("ENSG", "gene_id")) %>% 
  rename("Symbol" = "gene_id")
tpm.gene.df
```

### TPM FAM72

```{r}
tpm.gene.FAM72.df = 
  tpm.gene.df %>% 
  filter(str_detect(Symbol, "FAM72")) %>% 
  select(-ENSG)
tpm.gene.FAM72.df

# write.csv(x = tpm.gene.FAM72.df, 
#           file = paste0(data.dir, "wrangled_data/tpm_gene.FAM72.csv"), quote = F, row.names = F)
tpm.gene.FAM72.df = 
  read.csv(file = paste0(data.dir, "tpm_gene.FAM72.csv")) %>% 
  rename_with(.cols = starts_with("X"), .fn = ~ gsub("X", "", .x))
tpm.gene.FAM72.df
```

### write out estimated count for DESeq2

```{r}
# write.csv(x = est_count.gene.df,
#           file = paste0(data.dir, "wrangled_data/est_count.gene.df.csv"), quote = F, row.names = F)
est_count.gene.df = 
  read.csv(file = paste0(data.dir, "est_count.gene.df.csv")) %>% 
  rename_with(.cols = starts_with("X"), .fn = ~ gsub("X", "", .x))
est_count.gene.df

est_count.gene.FAM72.df = 
  est_count.gene.df %>% filter(str_detect(Symbol, "FAM72"))
# write.csv(x = est_count.gene.FAM72.df,
#           file = paste0(data.dir, "wrangled_data/est_count.gene.FAM72.csv"), quote = F, row.names = F)
```

# --------------------------------------------

# compare

## read in qPCR results

```{r}
fam72_qPCR.t = 
  read.csv(file = "/Users/minghan/GDrive_minghanpughlab/PughLabPMH/_projects/homolog_seq/data/fam72_qPCR.t.csv") %>% 
  filter(., apply(., 1, function(a_row) all(!is.na(a_row))))
# fam72_qPCR.t
```

## -------------------------------------------

## try TPM

### get replicate mean

```{r}
fam72_qPCR =
  as.tibble(t(fam72_qPCR.t %>% column_to_rownames(var="Samples")), rownames = "Symbol")
fam72_qPCR.mean = 
  fam72_qPCR %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA 231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))

TPM_gene.mean =
  tpm.gene.FAM72.df %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))
TPM_gene.mean

fam72_qPCR.mean.long = 
  fam72_qPCR.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "qPCR_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
# fam72_qPCR.mean.long

TPM_gene.mean.long = 
  TPM_gene.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "TPMgene_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
# TPM_gene.mean.long
```

### compare

#### raw values (good)

```{r}
fam72_qPCR_TPMgene_mean.joined = 
  left_join(fam72_qPCR.mean.long, TPM_gene.mean.long, by=c("Symbol_Cell_line", "Symbol", "Cell_line"))
fam72_qPCR_TPMgene_mean.joined

library(ggpubr)
```

##### all (R=0.88, p=1.6e-08)

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "Kallisto\nTPM vs qPCR FAM72 homolog expression correlation")

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCR_FAM72expr.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Cell_line'

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson", 
           label.x = 14000,
           label.x.npc = "middle", label.y.npc = "bottom") +
  labs(title = "Kallisto\nTPM vs qPCR FAM72 homolog expression correlation\nby cell line")
## p-value non-sig cuz too little samples

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCR_FAM72expr_byCellLine.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Symbol'

```{r}
fam72_qPCR_TPMgene_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "Kallisto\nTPM vs qPCR FAM72 homolog expression correlation\nby paralog")
## FAM72D not very good...

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCR_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

#### log2 qPCR (bad, 2 outliers)

- some say if qPCR is measuring doubling relative to reference (2^-ddCt method), 
    - value is essentially 'fold change'
    - should log2(fold change)

```{r}
fam72_qPCR_TPMgene_mean.joined.log2qPCR = 
  fam72_qPCR_TPMgene_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean))
fam72_qPCR_TPMgene_mean.joined.log2qPCR

library(ggpubr)

## all
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Cell_line'
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation\nby cell line")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Symbol'
fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation\nby paralog")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

#### log2 qPCR (2 outliers removed)

```{r}
fam72_qPCR_TPMgene_mean.joined.log2qPCR.outlier_removed = 
  fam72_qPCR_TPMgene_mean.joined.log2qPCR %>% 
  filter(Symbol_Cell_line != "FAM72B__HCT116_mean") %>% 
  filter(Symbol_Cell_line != "FAM72C__SW480_mean")
  
library(ggpubr)
```


```{r}
## all
fam72_qPCR_TPMgene_mean.joined.log2qPCR.outlier_removed %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation\noutliers removed (HCT116-FAM72B, SW480-FAM72C)")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_outlierRemoved_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Cell_line'
fam72_qPCR_TPMgene_mean.joined.log2qPCR.outlier_removed %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation\noutliers removed (HCT116-FAM72B, SW480-FAM72C)\nby cell line")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_outlierRemoved_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Symbol'
fam72_qPCR_TPMgene_mean.joined.log2qPCR.outlier_removed %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "Kallisto\nTPM vs log2(qPCR) FAM72 homolog expression correlation\noutliers removed (HCT116-FAM72B, SW480-FAM72C)\nby paralog")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMvsQPCRlog2_outlierRemoved_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

#### log2 both (not too bad)

- some also log2 both...
- Everaert,2017-Benchmarking of RNA-sequencing analysis workflows using wholetranscriptome RT-qPCR expression data

```{r}
fam72_qPCR_TPMgene_mean.joined.log2both = 
  fam72_qPCR_TPMgene_mean.joined %>% 
  mutate(qPCR_mean_log2 = log2(qPCR_mean)) %>% 
  mutate(TPMgene_mean_log2 = log2(TPMgene_mean))
fam72_qPCR_TPMgene_mean.joined.log2both

library(ggpubr)

## all
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "Kallisto\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMlog2vsQPCRlog2_FAM72expr.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Cell_line'
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson") +
  labs(title = "Kallisto\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation\nby cell line")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMlog2vsQPCRlog2_FAM72expr_byCellLine.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

## grouped by 'Symbol'
fam72_qPCR_TPMgene_mean.joined.log2both %>% 
  ggplot(aes(x=qPCR_mean_log2, y=TPMgene_mean_log2)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "Kallisto\nlog2(TPM) vs log2(qPCR) FAM72 homolog expression correlation\nby paralog")
ggsave(filename = paste0(data.dir, "figures/corr_kallisto_TPMlog2vsQPCRlog2_FAM72expr_byParalog.pdf"), 
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

## -------------------------------------------

## try DESeq2 normalized

- https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
- best for comparing a single gene, across samples

### DESeq2 normalize

```{r}
library(DESeq2)

## wrangle count data into matrix
expected_count.mat = 
  est_count.gene.df %>% 
  select(-ENSG) %>% 
  distinct(Symbol, .keep_all=T) %>% 
  column_to_rownames(var="Symbol") %>% 
  as.matrix() %>% 
  round()
expected_count.mat["FAM72A",]
expected_count.mat %>% dim() # [1] 59050    12

## make some fake 'conditions' and 'metadata'
condition = factor(c(rep("cell_line", 12)))
condition
metadata = data.frame(row.names=colnames(expected_count.mat), condition)
metadata

## Create DESeq2Dataset object
dds <- DESeqDataSetFromMatrix(countData = expected_count.mat, 
                              colData = metadata, 
                              design = ~ 1)
# View(counts(dds))

dds <- estimateSizeFactors(dds)
sizeFactors(dds)
#   293T_S44_L001   293T_S44_L002   A549_S43_L001   A549_S43_L002  Daudi_S41_L001  Daudi_S41_L002 
#       0.9864754       0.9942620       0.8934128       0.8914686       1.1065339       1.0993961 
# HCT116_S40_L001 HCT116_S40_L002 MDA231_S42_L001 MDA231_S42_L002  SW480_S39_L001  SW480_S39_L002 
#       0.9734714       0.9720464       1.0790530       1.0754216       1.0981376       1.0957488 

options(scipen = 999)   ## decimal
options(scipen = 0)     ## back to sci notation
DESeq2_normalized_counts <- counts(dds, normalized=TRUE)
DESeq2_normalized_counts

# write.table(DESeq2_normalized_counts,
#             file=paste0(data.dir,"wrangled_data/kallisto_DESeq2_normalized_counts.txt"),
#             sep="\t", quote=F, col.names=NA)

DESeq2_normalized_counts.FAM72 =
  DESeq2_normalized_counts %>% 
  as.data.frame %>% 
  rownames_to_column(var="Symbol") %>% 
  filter(str_detect(Symbol, "FAM72")) %>% 
  arrange(Symbol)
DESeq2_normalized_counts.FAM72

# write.table(DESeq2_normalized_counts.FAM72,
#             file=paste0(data.dir,"wrangled_data/kallisto_DESeq2_normalized_counts.FAM72.txt"),
#             sep="\t", quote=F, row.names=F)
```

### get replicate mean

```{r}
DESeq2norm.FAM72.mean = 
  DESeq2_normalized_counts.FAM72 %>% 
  mutate(X293_mean = rowMeans(dplyr::select(., starts_with("293")), na.rm = TRUE)) %>% 
  mutate(A549_mean = rowMeans(dplyr::select(., starts_with("A549")), na.rm = TRUE)) %>% 
  mutate(Daudi_mean = rowMeans(dplyr::select(., starts_with("Daudi")), na.rm = TRUE)) %>% 
  mutate(HCT116_mean = rowMeans(dplyr::select(., starts_with("HCT116")), na.rm = TRUE)) %>% 
  mutate(MDA231_mean = rowMeans(dplyr::select(., starts_with("MDA231")), na.rm = TRUE)) %>% 
  mutate(SW480_mean = rowMeans(dplyr::select(., starts_with("SW480")), na.rm = TRUE)) %>% 
  dplyr::select(Symbol, ends_with("mean"))
DESeq2norm.FAM72.mean

DESeq2norm.FAM72.mean.long = 
  DESeq2norm.FAM72.mean %>% 
  pivot_longer(cols = -Symbol, names_to = "Cell_line", values_to = "DESeq2norm_mean") %>% 
  unite(col = "Symbol_Cell_line", Symbol, Cell_line, sep = "__", remove = F)
DESeq2norm.FAM72.mean.long
```

### compare

#### raw values

```{r}
fam72_qPCR_DESeq2norm_mean.joined = 
  left_join(fam72_qPCR.mean.long, DESeq2norm.FAM72.mean.long, by=c("Symbol_Cell_line", "Symbol", "Cell_line"))
fam72_qPCR_DESeq2norm_mean.joined

library(ggpubr)
```

##### all (R=0.88, p=9.8e-09)

```{r}
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(method="lm", se=F, size=0.5) +
  stat_cor(method = "pearson") +
  labs(title = "Kallisto\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation")

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_DESeq2normvsQPCR_FAM72expr.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Cell_line'

```{r}
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean, color=Cell_line)) +
  geom_point(aes(shape=Symbol)) +
  geom_smooth(aes(group=Cell_line), method="lm", se=F, size=0.5) +
  stat_cor(aes(group=Cell_line), method = "pearson",
           label.x = 14000,
           label.x.npc = "middle", label.y.npc = "bottom") +
  labs(title = "Kallisto\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation\nby cell line")

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_DESeq2normvsQPCR_FAM72expr_byCellLine.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)
```

##### grouped by 'Symbol'

- suppose to be what DESeq2 is best at normalizing
    - a single gene, across samples

```{r}
fam72_qPCR_DESeq2norm_mean.joined %>% 
  ggplot(aes(x=qPCR_mean, y=DESeq2norm_mean)) +
  geom_point(aes(color=Cell_line, shape=Symbol)) +
  geom_smooth(aes(group=Symbol), method="lm", se=F, size=0.2, colour="grey") +
  stat_cor(aes(group=Symbol), method = "pearson") +
  labs(title = "Kallisto\nDESeq2 normalized vs qPCR FAM72 homolog expression correlation\nby paralog")

ggsave(filename = paste0(data.dir, "figures/corr_kallisto_DESeq2normvsQPCR_FAM72expr_byParalog.pdf"),
       device = "pdf", width = 6, height = 4, units = "in", dpi = 300)

```

<br>

# EOF